analogout.py

```diff
@@ -0,0 +1,224 @@
+    
+        with StringIO(wave_str) as f:
+            reader = csv.reader(f, delimiter=delim)
+            for i,row in enumerate(reader):
+                wave_arr[i,:] = row
+
+        return wave_arr
+
+
+    def str_to_bool(self, boolstr):
+        """ 
+        return True or False case-insensitively for a string 'true' or 'false'
+
+        Args: 
+            'boolstr': string to be converted; not case-sensitive
+        Return:
+            'boolean': True or False. 
+        """
+
+        boolstr = boolstr.lower()
+        if boolstr == "true":
+            return True
+        elif boolstr == "false":
+            return False
+        else:
+            print("Expected a string 'true' or 'false' but received {boolstr}")
+            raise
+
+
+    def load_xml(self, node):
+        """
+        Initialize AnalogOutput instance attributes with xml from CsPy
+
+        Expects node.tag == "AnalogOutput"
+
+        Args:
+            'node': type is ET.Element. tag should be "HSDIO"
+        """
+        
+        assert node.tag == "AnalogOutput"
+
+        for child in node: 
+
+            # not sure if this is necessary... could probably remove
+            if type(child) == ET.Element:
+
+                if child.tag == "enable":
+                    self.enable = self.str_to_bool(child.text)
+
+                elif child.tag == "physicalChannels":
+                    self.physicalChannels = child.text
+
+                elif child.tag == "minimum":
+                    self.minValue = float(child.text)
+
+                elif child.tag == "maximum":
+                    self.maxValue = float(child.text)
+
+                elif child.tag == "clockRate":
+                    self.sampleRate = float(child.text) # samples per second in LabVIEW
+
+                elif child.tag == "waveform":
+                    self.waveforms = self.wave_from_str(child.text)
+
+                elif child.tag == "waitForStartTrigger":
+                    self.startTrigger.waitForStartTrigger = self.str_to_bool(child.text)
+
+                elif child.tag == "exportStartTrigger":
+                    self.exportTrigger.exportStartTrigger = self.str_to_bool(child.text)
+
+                elif child.tag == "triggerSource":
+                    self.startTrigger.source = child.text
+
+                elif child.tag == "exportStartTriggerDestination":
+                    self.exportTrigger.outputTerminal = child.text # TODO implement the exportTrigger
+
+                elif child.tag == "triggerEdge": # TODO: make ao_edges the correct form
+                    try:
+                        self.startTrigger.edge = StartTrigger.nidaqmx_edges[child.text]
+                    except KeyError as e:
+                        # TODO: replace with logger
+                        print(f"Not a valid {child.tag} value {child.text} \n {e}")
+                        raise
+
+                # TODO: external clock could be a class as in LabVIEW. TBD.
```

Comment:
What does the labview class do? Sometimes they just store some number of variables, if that's the case, no need to make it a class. Dictionaries or rc get the job done if convenient.

Reply:
Yep, I used rc for this.